src/engine/naive_engine.cc

```diff
@@ -55,20 +69,41 @@ class NaiveEngine final : public Engine {
     opr->opr_name = opr_name;
     return opr;
   }
+
   void DeleteOperator(OprHandle op) override {
     NaiveOpr *opr = op->Cast<NaiveOpr>();
     delete opr;
   }
+
   void Push(OprHandle op, Context exec_ctx, int priority = 0, bool profiling = false) override {
     NaiveOpr *opr = op->Cast<NaiveOpr>();
-    this->PushAsync(opr->fn,
-                    exec_ctx,
-                    opr->const_vars,
-                    opr->mutable_vars,
-                    opr->prop,
-                    priority,
-                    PROFILER_MESSAGE(opr->opr_name));
+    opr->profiling = profiling;
+    this->PushAsync([&](RunContext ctx, CallbackOnComplete on_complete) {
+#if MXNET_USE_PROFILER
```

Comment:
@piiswrong  
* This is because in NativeEngine, the `real call op function` is not in ```Push()```(in ThreadedEngine, it is), but in ```PushAsync()```. so before we call op function, we should start profiler, but in ```PushAsync()```, there is no parameter `bool profiling` to tell engine whether we need to start profiler, `profiling` comes form ```Push()```, so start profiler there.   this is not the same as ThreadedEngine, because in ThreadedEngine, PushAsync() will call Push(), just the opposite.  
* when start profiler, we should stop it after the op is done, so if we do not create a callback, then it will use itself callback fuction https://github.com/tornadomeet/mxnet/blob/ca8b91482bb47b6ba84c92aa94aa6ef68f82900a/src/engine/naive_engine.cc#L114, this callback has no information(the second param is null) about the op, so we can not get the handle of the corresponding op, and can not stop the paramters.    

this design has been discussed with @ZihengJiang